src/applications/personalization/search-representative/config/search/searchRepresentativeWidget.jsx

```diff
@@ -1,49 +1,147 @@
+import React, { useEffect, useState } from 'react';
 import { connect } from 'react-redux';
 import { setData } from 'platform/forms-system/src/js/actions';
 import { fetchRepresentativeSearchResults } from '../../actions';
+import SearchRepresentativeResult from './searchRepresentativeResult';
+import Pagination from '@department-of-veterans-affairs/component-library/Pagination';
+import { chunk } from 'lodash';
 
 const SearchRepresentativeWidget = props => {
   const { loading, representatives, formData } = props;
+  const [pages, setPages] = useState(1);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [maxPerPage] = useState(3);
+  const [currentlyShowingData, setCurrentlyShowingData] = useState([]);
+  const [paginatedData, setPaginatedData] = useState(null);
+
+  // when the page loads, load the initial data set from props into the list
+  function handleLoadData() {
+    // Creates an array of arrays of the data passed in as props
+    // in this shape [[{}, {}, {}], [{}, {}, {}], [{}, {}, {}]]
+    const chunkedData = chunk(representatives, maxPerPage);
+    setCurrentlyShowingData(chunkedData[0]);
+    setPaginatedData(chunkedData);
+    setPages(chunkedData.length);
+  }
+
+  const handleClick = value => {
+    const updatedFormData = {
+      ...formData,
+      preferredRepresentative: value,
+    };
+    props.setData(updatedFormData);
+  };
+
+  const handleDataPagination = page => {
+    setCurrentlyShowingData(paginatedData[page - 1]);
+    setCurrentPage(page);
+  };
+
   useEffect(
     function() {
       props.fetchRepresentativeSearchResults();
+      handleLoadData();
     },
     [loading],
   );
-  function handleClick(value) {
-    const updatedFormData = {
-      ...formData,
-      preferredRepresentative: value,
-    };
-    props.setData(updatedFormData);
-  }
 
   if (loading) {
     return <div>Loading...</div>;
   } else if (representatives.length > 0) {
     return (
       <div>
-        {representatives.map(option => {
+        <p>
+          Please speak to the service organization or representative before you
+          send your request. You’ll need to mail to your nearest VA regional
+          office or to the service organization or representative that is going
+          to help you.
+        </p>
+        {currentlyShowingData?.map(option => {
           return (
-            <div
-              key={option.id}
-              className="vads-u-background-color--gray-lighter vads-u-padding--1p5 vads-u-margin-bottom--1"
-            >
-              <p className="vads-u-font-size--h3 vads-u-font-family--serif vads-u-font-weight--bold">
-                {option.name}
-              </p>
-              <p>{option.city}</p>
-
-              <button
-                onClick={() => handleClick(option.name)}
-                className="usa-button-secondary"
-              >
-                Choose this representative
-              </button>
-            </div>
+            <>
```

Comment:
I don't think this needs to be a fragment. React doesn't like it not having a key at the very least.

Reply:
Yeah there was more content surrounding this at one point, removed this.